src/Agent.Sdk/SecretMasking/LegacyLoggedSecretMasker.cs

```diff
@@ -0,0 +1,141 @@
+﻿using Agent.Sdk.SecretMasking;
+using Microsoft.TeamFoundation.DistributedTask.Logging;
+using ISecretMasker = Microsoft.TeamFoundation.DistributedTask.Logging.ISecretMasker;
+
+namespace Agent.Sdk.Util.SecretMasking;
+
+
+/// <summary>
+/// Extended secret masker service, that allows to log origins of secrets
+/// </summary>
+public class LegacyLoggedSecretMasker : ILoggedSecretMasker
```

Comment:
>LegacyLoggedSecretMasker [](http://example.com/codeflow?start=13&length=24)

very minor, feel free to won't fix.  More FYI than anything else.  I'm fine with "Legacy" prefix.

Standard naming convention for class versioning in .NET is to suffix with "2", "3", etc, rather than "Legacy" prefix.

[https://stackoverflow.com/questions/35776519/naming-convention-how-to-name-a-different-version-of-the-same-class](https://stackoverflow.com/questions/35776519/naming-convention-how-to-name-a-different-version-of-the-same-class)